crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -275,129 +279,172 @@ impl FromWorld for MeshPipeline {
-                            ViewClusterBindings::min_size_cluster_offsets_and_counts(
-                                clustered_forward_buffer_binding_type,
-                            ),
+                count: None,
+            },
+            // ClusterOffsetsAndCounts
+            BindGroupLayoutEntry {
+                binding: 8,
+                visibility: ShaderStages::FRAGMENT,
+                ty: BindingType::Buffer {
+                    ty: clustered_forward_buffer_binding_type,
+                    has_dynamic_offset: false,
+                    min_binding_size: Some(
+                        ViewClusterBindings::min_size_cluster_offsets_and_counts(
+                            clustered_forward_buffer_binding_type,
                         ),
-                    },
-                    count: None,
+                    ),
                 },
-                BindGroupLayoutEntry {
-                    binding: 9,
-                    visibility: ShaderStages::VERTEX_FRAGMENT,
-                    ty: BindingType::Buffer {
-                        ty: BufferBindingType::Uniform,
-                        has_dynamic_offset: false,
-                        min_binding_size: Some(GlobalsUniform::min_size()),
-                    },
-                    count: None,
+                count: None,
+            },
+            // Globals
+            BindGroupLayoutEntry {
+                binding: 9,
+                visibility: ShaderStages::VERTEX_FRAGMENT,
+                ty: BindingType::Buffer {
+                    ty: BufferBindingType::Uniform,
+                    has_dynamic_offset: false,
+                    min_binding_size: Some(GlobalsUniform::min_size()),
+                },
+                count: None,
+            },
+            // Depth texture
+            BindGroupLayoutEntry {
+                binding: 10,
+                visibility: ShaderStages::FRAGMENT,
+                ty: BindingType::Texture {
+                    multisampled: false,
+                    sample_type: TextureSampleType::Depth,
+                    view_dimension: TextureViewDimension::D2,
+                },
+                count: None,
+            },
+            // Normal texture
+            BindGroupLayoutEntry {
+                binding: 11,
+                visibility: ShaderStages::FRAGMENT,
+                ty: BindingType::Texture {
+                    multisampled: false,
+                    sample_type: TextureSampleType::Float { filterable: true },
+                    view_dimension: TextureViewDimension::D2,
                 },
-            ],
+                count: None,
+            },
+        ];
+
+        let view_layout = render_device.create_bind_group_layout(&BindGroupLayoutDescriptor {
             label: Some("mesh_view_layout"),
+            entries: &layout_entries,
         });
 
+        // Modify for multisampled
+        layout_entries[10].ty = BindingType::Texture {
```

Comment:
I don't love the way that this is mutating by index: seems like a very easy way to introduce bugs later. Can we branch up front?

Reply:
Yeah, I hate it too, I tried an alternative approache where the entries are given by a function and you pass it an argument to control it. The issue is that it's not a branch, we need both list of entries to exist. Technically my new approach is less ideal because it means recreating the whole layout_entries vec. I don't think it's expensive enough to be an isssue, but this is an annoying problem in general. At least now you can't mess up the index, but they are hardcoded in a ton of other places already which isn't ideal, but fixing this is way out of scope of this PR since this is an issue with the entire rendering code.